### PR TITLE
BUG: do not copy spatial partitions when None

### LINKED ISSUE #120: _Frame tries to copy spatial_partitions even when it is None.
```python 
import geopandas as gpd
import dask_geopandas as dgpd
from shapely.geometry import Point

gdf = gpd.GeoSeries([Point(0, 0), Point(1, 1)])
dgdf = dgpd.from_geopandas(gdf, npartitions=2)
dgdf.copy()

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-57-2e1e1df5b72e> in <module>
      5 gdf = gpd.GeoSeries([Point(0, 0), Point(1, 1)])
      6 dgdf = dgpd.from_geopandas(gdf, npartitions=2)
----> 7 dgdf.copy()

/opt/anaconda/envs/apollo/lib/python3.8/site-packages/dask_geopandas/core.py in copy(self)
    169         """
    170         self_copy = super().copy()
--> 171         self_copy.spatial_partitions = self.spatial_partitions.copy()
    172         return self_copy
    173 

AttributeError: 'NoneType' object has no attribute 'copy'
```

This should be an easy fix, i'll make a PR for it if everyone is happy with just checking if it's None before copying it.

### PR DESCRIPTION
**Pull request by bernardpazio**
check if spatial partitions is none before copying to fix #120

### CODE REVIEW

File: tests/test_core.py
Comment (jorisvandenbossche) [inline tests/test_core.py:453]:
```suggestion
    assert ddf.spatial_partitions is None
    ddf_copy = ddf.copy()
    assert ddf_copy.spatial_partitions is None

```

(not strictly needed, but making it a bit more explicit)

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks for the PR! Added a small comment, otherwise looks good

File: tests/test_core.py
Comment (bernardpazio) [reply tests/test_core.py]: yeah that is fair, I've gone with explicitly setting it to None, just in case something happens in between which makes it not None at some point.

**Review by martinfleis — APPROVED** [review]
Thanks!

(let's wait for #119 to make CI happy before merging)

**Comment by jorisvandenbossche** [comment]
Thanks again @bernardpazio !

**Pull request merged**
